src/kmergraph.cpp

```diff
@@ -238,20 +218,271 @@ void KmerGraph::discover_k() {
+                       assert_msg(stoi(split_line[1]) << ">=" << outnode_counts.size()));
+                assert(stoi(split_line[3]) < (int) innode_counts.size() or
+                       assert_msg(stoi(split_line[3]) << ">=" << innode_counts.size()));
+                outnode_counts[stoi(split_line[1])] += 1;
+                innode_counts[stoi(split_line[3])] += 1;
+            }
+        }
+
+        if (id == 0) {
+            reverse(nodes.begin(), nodes.end());
+        }
+
+        id = 0;
+        for (const auto &n : nodes) {
+            assert(nodes[id]->id == id);
+            id++;
+            assert(n->id < outnode_counts.size() or assert_msg(n->id << ">=" << outnode_counts.size()));
+            assert(n->id < innode_counts.size() or assert_msg(n->id << ">=" << innode_counts.size()));
+            n->outNodes.reserve(outnode_counts[n->id]);
+            n->inNodes.reserve(innode_counts[n->id]);
+        }
+
+        myfile.clear();
+        myfile.seekg(0, myfile.beg);
+
+        while (getline(myfile, line).good()) {
+            if (line[0] == 'L') {
+                split_line = split(line, "\t");
+                assert(split_line.size() >= 5);
+                if (split_line[2] == split_line[4]) {
+                    from = std::stoi(split_line[1]);
+                    to = std::stoi(split_line[3]);
+                } else {
+                    // never happens
+                    from = std::stoi(split_line[3]);
+                    to = std::stoi(split_line[1]);
+                }
+                add_edge(nodes[from], nodes[to]);
+                //nodes[from]->outNodes.push_back(nodes.at(to));
+                //nodes[to]->inNodes.push_back(nodes.at(from));
+            }
+        }
+    } else {
+        std::cerr << "Unable to open kmergraph file " << filepath << std::endl;
+        exit(1);
+    }
+}
+
+
+uint32_t KmerGraph::min_path_length() {
+    //TODO: FIX THIS INNEFICIENCY I INTRODUCED
+    std::vector<KmerNodePtr> sorted_nodes(this->sorted_nodes.begin(), this->sorted_nodes.end());
+
+    if (shortest_path_length > 0) {
+        return shortest_path_length;
+    }
+
+#ifndef NDEBUG
+    //TODO: check if tests must be updated or not due to this (I think not - sorted_nodes is always sorted)
+    //if this is added, some tests bug, since it was not executed before...
+    //check();
+#endif
+
+    std::vector<uint32_t> len(sorted_nodes.size(), 0); // length of shortest path from node i to end of graph
+    for (uint32_t j = sorted_nodes.size() - 1; j != 0; --j) {
+        for (uint32_t i = 0; i != sorted_nodes[j - 1]->outNodes.size(); ++i) {
+            if (len[sorted_nodes[j - 1]->outNodes[i].lock()->id] + 1 > len[j - 1]) {
+                len[j - 1] = len[sorted_nodes[j - 1]->outNodes[i].lock()->id] + 1;
+            }
+        }
+    }
+    shortest_path_length = len[0];
+    return len[0];
+}
+
+bool KmerGraph::operator==(const KmerGraph &y) const {
```

Comment:
Suggest rename of `y` to `other_graph`

Reply:
Done in https://github.com/leoisl/pandora/commit/0de22cf8f37349e8cedb4815dc95b51e715866ce